openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,270 @@
+
+
+class Signatory(TimeStampedModel):
+    """
+    Signatory model to add certificate signatories.
+    """
+    name = models.CharField(max_length=255)
+    title = models.CharField(max_length=255)
+    image = models.ImageField(
+        help_text=_(
+            u"Image must be square PNG files. The file size should be under 250KB."
+        ),
+        upload_to=assets_path,
+        validators=[validate_image]
+    )
+
+    def __unicode__(self):
+        """Unicode representation. """
+        return u"{name}, {title}".format(
+            name=self.name,
+            title=self.title,
+        )
+
+    def save(self, *args, **kwargs):
+        """save the certificate template asset """
+        if self.pk is None:
+            temp_image = self.image
+            self.image = None
+            super(Signatory, self).save(*args, **kwargs)
+            self.image = temp_image
+
+        super(Signatory, self).save(*args, **kwargs)
+
+
+class CertificateTemplate(TimeStampedModel):
+    """
+    Certificate Template model to organize content for certificates.
+    """
+
+    name = models.CharField(max_length=255, db_index=True)
+    content = models.TextField(
+        help_text=_(u"Template content data.")
+    )
+    certificate_type = models.CharField(
+        max_length=32, null=True, blank=True,
+        choices=MODE_CHOICES,
+    )
+    organization_id = models.IntegerField(
+        null=True,
+        blank=True,
+        db_index=True,
+        help_text=_(u'Organization of template.'),
+    )
+
+    def __unicode__(self):
+        """Unicode representation. """
+        return u"{name}".format(
+            name=self.name
+        )
+
+
+class AbstractCertificate(AbstractCredential):
+    """
+    Abstract Certificate configuration to support multiple type of certificates
+    i.e. Programs, Courses.
+    """
+    signatory = models.ForeignKey(Signatory)
+    template = models.ForeignKey(CertificateTemplate, null=True, blank=True)
+    title = models.CharField(max_length=255, null=True, blank=True)
+
+
+class CourseCertificate(AbstractCertificate):
+    """
+    Configuration for Course Certificates.
+    """
+    course_id = CourseKeyField(max_length=255)
+    certificate_type = models.CharField(
+        max_length=32,
+        choices=MODE_CHOICES
+    )
+
+    class Meta(object):
+        # Enforce the constraint that mode and certificate should be unique
+        # together.
+        unique_together = (('course_id', 'certificate_type'),)
+
+    def __unicode__(self):
+        """Unicode representation. """
+        return u"{course_id}, {certificate_type}".format(
+            course_id=self.course_id,
+            certificate_type=self.certificate_type,
+        )
+
+
+class ProgramCertificate(AbstractCertificate):
+    """
+    Configuration for Program Certificates.
+    """
+    program_id = models.IntegerField(
+        db_index=True,
+        unique=True,
+        help_text=_(u'Programs Id.'),
```

Comment:
This help text isn't helpful as it doesn't tell us anything we don't already know from the field name. Remove it.